libethereum/ChainParams.cpp

```diff
@@ -49,44 +49,67 @@ ChainParams::ChainParams()
 
 ChainParams::ChainParams(std::string const& _json, h256 const& _stateRoot)
 {
+	try
+	{
+		loadConfig(_json, _stateRoot);
+	}
+	catch (...)
+	{
+		cerr << "Bad json file provided. Should be a config file" << endl;
```

Comment:
Please leak the exception - it should not be possible to construct a ChainParams object with an invalid config file. Furthermore, we should not assume that `cerr` is available / read here.